Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -1112,6 +1172,23 @@ class MultiTrajectory {
   template <typename F>
   void visitBackwards(IndexType iendpoint, F&& callable) const;
 
+  /// Range for the track states from @p iendpoint to the trajectory start
+  /// @param iendpoint Trajectory entry point to start from
+  /// @return Iterator pair to iterate over
+  /// @note Const version
+  auto trackStateRange(IndexType iendpoint) const {
+    return detail_lt::TrackStateRange{getTrackState(iendpoint)};
+  }
+
+  /// Range for the track states from @p iendpoint to the trajectory start
+  /// @param iendpoint Trajectory entry point to start from
+  /// @return Iterator pair to iterate over
+  /// @note Mutable version
+  template <bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
+  auto trackStateRange(IndexType iendpoint) {
```

Comment:
Is it possible to get a const `TrackStateRange` from a mutable trajectory? I think there is always the mutable overload called.
I would actually prefer two different function names `trackStateRange` and `constTrackStateRange`.

Reply:
Well I was thinking in that case you would write something like

```cpp
for(const auto ts : mtj.trackStateRange()) {}
```

Do you think that's not enough?

Reply:
For the for loop, this works, you're right. But I also thought that users might want to write or use algorithms in terms of ranges (e.g., C++20 ranges algorithms...). 
Therefore I would found it nice to have this options...